src/main/java/com/commercetools/project/sync/shoppinglist/ShoppingListSyncer.java

```diff
@@ -0,0 +1,101 @@
+import io.sphere.sdk.shoppinglists.ShoppingListDraft;
+import io.sphere.sdk.shoppinglists.queries.ShoppingListQuery;
+import java.time.Clock;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ShoppingListSyncer
+    extends Syncer<
+        ShoppingList,
+        ShoppingListDraft,
+        ShoppingListSyncStatistics,
+        ShoppingListSyncOptions,
+        ShoppingListQuery,
+        ShoppingListSync> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ShoppingListSyncer.class);
+
+  private ShoppingListSyncer(
+      @Nonnull ShoppingListSync sync,
+      @Nonnull SphereClient sourceClient,
+      @Nonnull SphereClient targetClient,
+      @Nonnull CustomObjectService customObjectService,
+      @Nonnull Clock clock) {
+    super(sync, sourceClient, targetClient, customObjectService, clock);
+  }
+
+  public static ShoppingListSyncer of(
+      @Nonnull final SphereClient sourceClient,
+      @Nonnull final SphereClient targetClient,
+      @Nonnull final Clock clock) {
+    final QuadConsumer<
+            SyncException,
+            Optional<ShoppingListDraft>,
+            Optional<ShoppingList>,
+            List<UpdateAction<ShoppingList>>>
+        logErrorCallback =
+            (exception, newResourceDraft, oldResource, updateActions) ->
+                logErrorCallback(
+                    LOGGER,
+                    "shoppingList",
+                    exception,
+                    oldResource.map(ShoppingList::getKey).orElse(IDENTIFIER_NOT_PRESENT),
+                    updateActions);
+    final TriConsumer<SyncException, Optional<ShoppingListDraft>, Optional<ShoppingList>>
+        logWarningCallback =
+            (exception, newResourceDraft, oldResource) ->
+                logWarningCallback(
+                    LOGGER,
+                    "shoppingList",
+                    exception,
+                    oldResource.map(ShoppingList::getKey).orElse(IDENTIFIER_NOT_PRESENT));
+    final ShoppingListSyncOptions shoppingListSyncOptions =
+        ShoppingListSyncOptionsBuilder.of(targetClient)
+            .errorCallback(logErrorCallback)
+            .warningCallback(logWarningCallback)
+            .build();
+
+    final ShoppingListSync shoppingListSync = new ShoppingListSync(shoppingListSyncOptions);
+
+    final CustomObjectService customObjectService = new CustomObjectServiceImpl(targetClient);
+
+    return new ShoppingListSyncer(
+        shoppingListSync, sourceClient, targetClient, customObjectService, clock);
+  }
+
+  @Nonnull
+  @Override
+  protected CompletionStage<List<ShoppingListDraft>> transform(@Nonnull List<ShoppingList> page) {
+    return CompletableFuture.completedFuture(
+        ShoppingListReferenceResolutionUtils.mapToShoppingListDrafts(page));
```

Comment:
I would favor static imports over the static method calls.

Reply:
Done, https://github.com/commercetools/commercetools-project-sync/pull/200/commits/d765aab2b758e62b59e8dadeadde87ae0954c30b